control/grpc.py

```diff
@@ -1983,14 +2064,53 @@ def list_subsystems_safe(self, request, context):
                     s["namespace_count"] = 0
                     s["enable_ha"] = False
                 # Parse the JSON dictionary into the protobuf message
-                subsystem = pb2.subsystem()
+                subsystem = pb2.subsystem_cli()
                 json_format.Parse(json.dumps(s), subsystem, ignore_unknown_fields=True)
                 subsystems.append(subsystem)
             except Exception:
                 self.logger.exception(f"{s=} parse error: ")
                 pass
 
-        return pb2.subsystems_info(status = 0, error_message = os.strerror(0), subsystems=subsystems)
+        return pb2.subsystems_info_cli(status = 0, error_message = os.strerror(0), subsystems=subsystems)
+
+    def get_subsystems_safe(self, request, context):
+        """Gets subsystems."""
+
+        self.logger.info(f"Received request to get subsystems, context: {context}")
+        subsystems = []
+        try:
+            ret = rpc_nvmf.nvmf_get_subsystems(self.spdk_rpc_client)
+            self.logger.info(f"get_subsystems: {ret}")
+        except Exception as ex:
+            self.logger.error(f"get_subsystems failed with: \n {ex}")
+            context.set_code(grpc.StatusCode.INTERNAL)
```

Comment:
this method is not used by CLI rather by c++ binary, using gRPC native error return convention seems reasonable. wdyt?

Reply:
When we changed the status codes it was decided to use errno values. Not that I found this decision that good as most of the times I didn't have a specific errno error I could use and just used EINVAL. But, as this is what we use all over the place it might be confusing if we mix it with gRPC error statuses.